app/src/lib/stores/app-store.ts

```diff
@@ -1023,18 +1023,49 @@ export class AppStore {
     return null
   }
 
-  /** Update the menu with the names of the user's preferred apps. */
-  private updatePreferredAppMenuItemLabels() {
+  /**
+   * Update menu labels for editor, shell, and pull requests.
+   */
+  private async updateMenuItemLabels(repository?: Repository) {
     const editorLabel = this.selectedExternalEditor
       ? `Open in ${this.selectedExternalEditor}`
       : undefined
 
+    let prLabel: string | undefined
+
+    if (repository) {
+      prLabel = await this.getPullRequestLabel(repository)
+    }
+
     updatePreferredAppMenuItemLabels({
       editor: editorLabel,
+      pullRequestLabel: prLabel,
       shell: `Open in ${this.selectedShell}`,
     })
   }
 
+  private getPullRequestLabel(repository: Repository) {
+    const githubRepository = repository.gitHubRepository
+
+    if (!githubRepository) {
+      return
+    }
+
+    const repositoryState = this.repositoryState.get(repository.hash)
+
+    if (!repositoryState) {
+      return
+    }
+
+    const branchState = repositoryState.branchesState
+
+    if (!branchState.currentPullRequest) {
+      return
```

Comment:
Should we return "Create Pull Request" here? That way we're consistent in how we determine the label for the pull request menu item.

Reply:
We could, but I followed the pattern of having a default value in the function signature. If I return a string here, it would be doing the same thing that happens now. 

https://github.com/desktop/desktop/blob/2b881e67083d08989d018671238eec23afb1ef1f/app/src/main-process/menu/build-default-menu.ts#L16-L24

I personally prefer returning a string vs undefined.

Reply:
>I personally prefer returning a string vs undefined.

Agree, let's do it 👍